### PR TITLE
GH-540, add `federated_swift_repository` resource, 

### PR DESCRIPTION
**Pull request by danielmkn**
Also added tests and documentation.
Fix federated test, now it works correctly with terraform repos.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 6.17.0 (October 21, 2022). Tested on Artifactory 7.46.6
```
Comment (Turhsus) [inline CHANGELOG.md:1]: Is this semantic versioning?

Just out of curiosity, how do you decide to go from 6.16.4 -> 6.17.0 vs 6.16.5

Does this PR count as a minor feature addition?

File: pkg/artifactory/resource/repository/federated/resource_artifactory_federated_repository_test.go [hunk 2]
```diff
@@ -100,6 +101,9 @@ func federatedTestCase(repoType string, t *testing.T) (*testing.T, resource.Test
 		"xrayIndex":    xrayIndex,
 		"memberUrl":    federatedMemberUrl,
 	}
+
+	repoTypeAdjusted := local.GetPackageType(repoType)
+
```
Comment (shrajfr12) [inline pkg/artifactory/resource/repository/federated/resource_artifactory_federated_repository_test.go:106]: What if the repoType is not one of the supported types ? Any special handling to indicate Not a supported type ?

File: CHANGELOG.md (hunk 1 above)
Comment (danielmkn) [reply CHANGELOG.md:1]: [Terraform guidance.](https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/versioning#versioning-specification)

File: docs/resources/federated_swift_repository.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+## Argument Reference
+
+Arguments have a one to one mapping with the [JFrog API](https://www.jfrog.com/confluence/display/JFROG/Repository+Configuration+JSON#RepositoryConfigurationJSON-FederatedRepository).
+
+The following arguments are supported, along with the [common list of arguments from the local repositories](local.md):
+
+* `key` - (Required) the identity key of the repo.
+* `member` - (Required) The list of Federated members and must contain this repository URL (configured base URL
+  `/artifactory/` + repo `key`). Note that each of the federated members will need to have a base URL set.
+  Please follow the [instruction](https://www.jfrog.com/confluence/display/JFROG/Working+with+Federated+Repositories#WorkingwithFederatedRepositories-SettingUpaFederatedRepository)
```
Comment (Turhsus) [inline docs/resources/federated_swift_repository.md:35]: nit: "Please follow the instructions"

File: pkg/artifactory/resource/repository/federated/resource_artifactory_federated_repository_test.go (hunk 2 above)
Comment (danielmkn) [reply pkg/artifactory/resource/repository/federated/resource_artifactory_federated_repository_test.go:106]: The verification for the not supported types is in the schema. The provider won't take the HCL with the type, which is not in the list of supported repos. It's not done in the test here.

File: docs/resources/federated_swift_repository.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+subcategory: "Federated Repositories"
+---
+# Artifactory Federated Swift Repository Resource
+
+Creates a federated Swift repository.
+
+## Example Usage
```
Comment (Turhsus) [inline docs/resources/federated_swift_repository.md:8]: This is good documentation IMO

**Review by Turhsus — APPROVED** [review]
LGTM

File: pkg/artifactory/resource/repository/federated/resource_artifactory_federated_repository_test.go (hunk 2 above)
Comment (shrajfr12) [reply pkg/artifactory/resource/repository/federated/resource_artifactory_federated_repository_test.go:106]: ok

**Review by shrajfr12 — APPROVED** [review]

**Review by mritunjaykumar — COMMENTED** [review]
LGTM

**Pull request merged**